src/types/dict.rs

```diff
@@ -430,7 +438,9 @@ enum DictIterImpl {
 
 impl DictIterImpl {
     #[inline]
-    fn next<'py>(
+    /// Safety: the dict should be locked with a critical section on the free-threaded build
+    /// and otherwise not shared between threads in code that releases the GIL.
```

Comment:
```suggestion
    /// Safety: the dict should be locked with a critical section on the free-threaded build
    /// and otherwise not shared between threads in code that releases the GIL.
    #[inline]
```